config/confighttp/confighttp.go

```diff
@@ -32,6 +33,28 @@ import (
 	"go.opentelemetry.io/collector/internal/middleware"
 )
 
+type CompressionType string
+
+const (
+	CompressionUnsupported CompressionType = ""
+	CompressionGzip        CompressionType = "gzip"
+	CompressionZlib        CompressionType = "zlib"
+	CompressionDeflate     CompressionType = "deflate"
+	CompressionSnappy      CompressionType = "snappy"
+	CompressionZstd        CompressionType = "zstd"
```

Comment:
```suggestion
	CompressionZstd        CompressionType = "zstd"
	CompressionNone      CompressionType = "none"
```

There should be an explicit option for specifying no compression.

Reply:
done